dev/com.ibm.ws.jsf.2.2_fat/test-applications/TestJSFEL.war/src/com/ibm/ws/jsf22/el/components/CustomComponent.java

```diff
@@ -108,42 +156,72 @@ private ELResolver[] getELResolvers(ELResolver elResolver, ResponseWriter respon
-                add("class javax.el.MapELResolver");
-                add("class javax.el.ResourceBundleELResolver");
-                add("class javax.el.ListELResolver");
-                add("class javax.el.ArrayELResolver");
-                add("class javax.el.BeanELResolver");
+                if (isFaces40OrLater) {
+                    add(expFactory.getStreamELResolver().getClass().getName());
+                    add("javax.el.StaticFieldELResolver");
+                    add("javax.el.MapELResolver");
+                    add("javax.el.ListELResolver");
+                    add("javax.el.ArrayELResolver");
+                    add("org.apache.myfaces.el.resolver.LambdaBeanELResolver");
+                } else {
+                    add(expFactory.getStreamELResolver().getClass().getName());
+                    add("javax.el.StaticFieldELResolver");
+                    add("javax.el.MapELResolver");
+                    add("javax.el.ListELResolver");
+                    add("javax.el.ArrayELResolver");
+                    add("javax.el.BeanELResolver");
+                }
             }
         };
 
+        // Loop over all the resolvers until we find the StreamELResolver and then
+        // copy the next 5 resolvers into a new array to use for comparison.
+        int firstResolverFoundIndex = 0;
+        List<String> actualResolvers = new ArrayList<String>();
+        for (int i = 0; i < resolvers.length; i++) {
+            if (resolvers[i].getClass().getName().equals(expectedResolvers.get(0))) {
+                firstResolverFoundIndex = i;
+                break;
+            }
+        }
+
+        // Add all the actual resolvers to the actualResolvers list.
+        for (int i = firstResolverFoundIndex; i <= firstResolverFoundIndex + 5; i++) {
+            actualResolvers.add(resolvers[i].getClass().getName());
+        }
+
+        // Compare the two lists to ensure they have the expected resolvers in the expected order.
         int elResolversCounter = 0;
-        if (resolvers.length != 0) {
-            for (int i = 0; i < resolvers.length; i++) {
-                if (resolvers[i] != null) {
-                    responseWriter.write(resolvers[i].getClass().toString());
+        if (actualResolvers.size() != 0) {
+            String resolver;
+            for (int i = 0; i < actualResolvers.size(); i++) {
+                resolver = actualResolvers.get(i);
+                if (resolver != null) {
+                    responseWriter.write(resolver);
                     elResolversCounter++;
-                    if (!resolvers[i].getClass().toString().equals(expectedResolvers.get(i))) {
+                    if (!resolver.equals(expectedResolvers.get(i))) {
                         return false;
                     }
                 }
             }
+
             if (elResolversCounter != expectedResolvers.size()) {
                 return false;
             }
```

Comment:
Small thing:  I would move this size check before the list comparison.

Reply:
Yea, that makes sense let me get a change up for this. Thanks!

Reply:
Need to compare and loop over the resolvers before we know the number of Resolvers found.